core/src/com/unciv/logic/map/tile/TileInfoImprovementFunctions.kt

```diff
@@ -223,6 +225,26 @@ class TileInfoImprovementFunctions(val tile: Tile) {
         )
             takeOverTilesAround(civToActivateBroaderEffects, tile)
 
+        if (civToActivateBroaderEffects != null && improvementObject != null) {
+            for (unique in improvementObject.uniqueObjects)
+                if (!unique.hasTriggerConditional()) {
```

Comment:
Please add this 'if' as .uniqueObjects.filter{} instead, this decision tree is too long